dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/internal/ApplicationRegistryImpl.java

```diff
@@ -66,13 +67,33 @@ public class ApplicationRegistryImpl implements ApplicationRegistry {
     @Reference
     private MergeProcessor mergeProcessor;
 
+    @SuppressWarnings("unused")
+    private OpenAPIAppConfigProvider openAPIAppConfigProvider;
+
+    @Reference(cardinality = ReferenceCardinality.MANDATORY, policy = ReferencePolicy.STATIC, unbind = "unbindAppConfigListener")
+    public void bindAppConfigListener(OpenAPIAppConfigProvider openAPIAppConfigProvider) {
+        this.openAPIAppConfigProvider = openAPIAppConfigProvider;
+        openAPIAppConfigProvider.registerAppConfigListener(this);
+    }
+
+    public void unbindAppConfigListener(OpenAPIAppConfigProvider openAPIAppConfigProvider) {
+        openAPIAppConfigProvider.unregisterAppConfigListener(this);
+        this.openAPIAppConfigProvider = null;
+    }
+
+    @Reference
+    public void bindOpenAPIAppConfigProvider(OpenAPIAppConfigProvider openAPIAppConfigProvider) {
+        this.openAPIAppConfigProvider = null;
+        openAPIAppConfigProvider.registerAppConfigListener(this);
+    }
```

Comment:
This looks like you have two references to `OpenAPIAppConfigProvider`, but the first one is named `appConfigListener`.